Added feature that generates the elementary weight of a rooted tree as a latexstring

pw0lf opened the pull request:
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API
  
## Additional context

Added the feature as described in #96.
Also added test and they all passed.
Added documentation, but the picture of the output is still missing, since I don't remember how to add pictures there ...
This picture should be displayed:
![Bildschirmfoto 2024-01-20 um 23 07 43](https://github.com/SciML/RootedTrees.jl/assets/125130707/8a035faf-fd1a-4fc0-92be-c3387eb53177)

pw0lf commented:
Almost forgot: 
Because a function called `elementary_weight` already existed, I called the function `elementary_weight_latexstring`.
Should I rename `elementary_differential` to `elementary_differential_latexstring` out of consistency?

ranocha commented:
> Almost forgot: Because a function called `elementary_weight` already existed, I called the function `elementary_weight_latexstring`. Should I rename `elementary_differential` to `elementary_differential_latexstring` out of consistency?

Good point 👍 Could you please rename the function as suggested - and deprecate the old one?

```julia
help?> @deprecate
  @deprecate old new [export_old=true]

  Deprecate method old and specify the replacement call new, defining a new method old with the
  specified signature in the process.

  To prevent old from being exported, set export_old to false.

  │ Julia 1.5
  │
  │  As of Julia 1.5, functions defined by @deprecate do not print warning when julia is
  │  run without the --depwarn=yes flag set, as the default value of --depwarn option is
  │  no. The warnings are printed from tests run by Pkg.test().

  Examples
  ≡≡≡≡≡≡≡≡

  julia> @deprecate old(x) new(x)
  old (generic function with 1 method)
```

pw0lf commented:
I changed the name and deprecated the old function. Also changed the version number to 3.0.0 since it is a breaking change, right?

Also the picture should be added to the documentation and I don't know how...
Could you help me with that, @ranocha?

ranocha commented on Project.toml:
```diff
@@ -1,7 +1,7 @@
 uuid = "47965b36-3f3e-11e9-0dcf-4570dfd42a8c"
 authors = ["Hendrik Ranocha <mail@ranocha.de> and contributors"]
-version = "2.20.0"
+version = "3.0.0"
```
```suggestion
version = "2.21.0"
```
It's not a breaking change to deprecate something. Deprecating the old function means that it's still accessible. It just informs people that they should use another function in the future (if deprecation warnings are turned on).

ranocha commented on src/RootedTrees.jl:
```diff
@@ -17,7 +17,8 @@ end
        ColoredRootedTree, BicoloredRootedTree, BicoloredRootedTreeIterator
 
-export butcher_representation, elementary_differential
+export butcher_representation, elementary_differential, elementary_differential_latexstring,
+       elementary_weight_latexstring
```
```suggestion
export butcher_representation, elementary_differential_latexstring,
       elementary_weight_latexstring
```
We shouldn't need to `export` the deprecated name.

ranocha commented:
> Also the picture should be added to the documentation and I don't know how...

You can insert some code similar to

https://github.com/SciML/RootedTrees.jl/blob/e9698cffe6bc5c5bae0d7a0e09c52928e8639e28/docs/src/tutorials/basics.md?plain=1#L83-L85

In this case, the link to your picture above is https://private-user-images.githubusercontent.com/125130707/298310491-8a035faf-fd1a-4fc0-92be-c3387eb53177.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxNjY1ODAsIm5iZiI6MTcwNjE2NjI4MCwicGF0aCI6Ii8xMjUxMzA3MDcvMjk4MzEwNDkxLThhMDM1ZmFmLWZkMWEtNGZjMC05MmJlLWMzMzg3ZWI1MzE3Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyNVQwNzA0NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzQyYjM0MWY0MDU5YWZlYzBmODA4MjFiZGIxN2E3YjhkYTdmZDNkYTU5NmI5OTEwNWFiZjg0OGZjNDg1MzZhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GB-PigOlQqenzgruzWg19qslzM6RXeX4xWwCNreOvNY

ranocha commented:
The code is something like

```markdown
![latex elemenary weights](https://private-user-images.githubusercontent.com/125130707/298310491-8a035faf-fd1a-4fc0-92be-c3387eb53177.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxNjY1ODAsIm5iZiI6MTcwNjE2NjI4MCwicGF0aCI6Ii8xMjUxMzA3MDcvMjk4MzEwNDkxLThhMDM1ZmFmLWZkMWEtNGZjMC05MmJlLWMzMzg3ZWI1MzE3Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyNVQwNzA0NDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzQyYjM0MWY0MDU5YWZlYzBmODA4MjFiZGIxN2E3YjhkYTdmZDNkYTU5NmI5OTEwNWFiZjg0OGZjNDg1MzZhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GB-PigOlQqenzgruzWg19qslzM6RXeX4xWwCNreOvNY)
```

ranocha reviewed (approved):
Thanks a lot! It's weird that docs CI fails. I will check it in another clean PR (https://github.com/SciML/RootedTrees.jl/pull/157)
